core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -68,22 +75,23 @@ impl DataAvailabilityDispatcher {
                 .max_concurrent_requests
                 .unwrap_or(DEFAULT_MAX_CONCURRENT_REQUESTS) as usize,
         );
+        let (shutdown_tx, mut shutdown_rx) = watch::channel(false);
 
         let next_expected_batch = Arc::new(Mutex::new(None));
 
         let stop_receiver_clone = stop_receiver.clone();
         let pool_clone = self.pool.clone();
         let config_clone = self.config.clone();
         let next_expected_batch_clone = next_expected_batch.clone();
-        let pending_blobs_reader = tokio::spawn(async move {
+        let mut pending_blobs = JoinSet::new();
+        pending_blobs.spawn(async move {
```

Comment:
Considering that we now don't have a `_reader` variable, add a comment above this line stating that this task **reads** pending blobs from de database.